lib/webResource.ts

```diff
@@ -9,6 +9,22 @@ import { generateUuid } from "./util/utils";
 export type HttpMethods = "GET" | "PUT" | "POST" | "DELETE" | "PATCH" | "HEAD" | "OPTIONS" | "TRACE";
 export type HttpRequestBody = Blob | string | ArrayBuffer | ArrayBufferView | (() => NodeJS.ReadableStream);
 
+/**
+ * Fired in response to upload or download progress (browser only).
+ */
+export type TransferProgressEvent = {
+  /**
+   * The number of bytes loaded so far.
+   */
+  loaded: number,
```

Comment:
Were there any other names you considered besides `loaded`?

Reply:
I just used the names from the underlying [ProgressEvent](https://developer.mozilla.org/en-US/docs/Web/API/ProgressEvent).

Reply:
So `loadedBytes` and `totalBytes` might be better?

Reply:
I prefer those names. They remind me that we're dealing with bytes, without having to lookup the documentation.